File: MorWalPizVideo.Models/Models/ApiKey.cs
using MongoDB.Bson.Serialization.Attributes;
using MorWalPizVideo.Server.Models;

namespace MorWalPizVideo.Models.Models;

public record ApiKey : BaseEntity
{
    [BsonElement("key")]
    public string Key { get; init; } = string.Empty;

    [BsonElement("name")]
    public string Name { get; init; } = string.Empty;

    [BsonElement("description")]
    public string Description { get; init; } = string.Empty;

    [BsonElement("isActive")]
    public bool IsActive { get; init; } = true;

    [BsonElement("lastUsedAt")]
    public DateTime? LastUsedAt { get; init; }

    [BsonElement("expiresAt")]
    public DateTime? ExpiresAt { get; init; }

    [BsonElement("rateLimitPerMinute")]
    public int RateLimitPerMinute { get; init; } = 60;

    [BsonElement("allowedIpAddresses")]
    public List<string> AllowedIpAddresses { get; init; } = new();
}

File: frontend/back-office-spa/src/models/apiKey.ts
/**
 * API Key TypeScript models
 * Maps to DTOs from MorWalPizVideo.BackOffice/Controllers/ApiKeysController.cs
 */

export interface ApiKeyDto {
  id: string;
  name: string;
  description: string;
  isActive: boolean;
  rateLimitPerMinute: number;
  allowedIpAddresses: string[];
  lastUsedAt: string | null;
  expiresAt: string | null;
  createdAt: string;
}

export interface CreateApiKeyRequest {
  name: string;
  description?: string;
  rateLimitPerMinute?: number;
  allowedIpAddresses?: string[];
  expiresAt?: string;
}

export interface CreateApiKeyResponse {
  id: string;
  name: string;
  description: string;
  key: string; // ONLY returned once on creation
  rateLimitPerMinute: number;
  allowedIpAddresses: string[];
  expiresAt: string | null;
  createdAt: string;
  message: string;
}

export interface UpdateApiKeyRequest {
  name?: string;
  description?: string;
  rateLimitPerMinute?: number;
  allowedIpAddresses?: string[];
  expiresAt?: string;
}

export interface ToggleApiKeyResponse {
  message: string;
  isActive: boolean;
}

export interface RegenerateApiKeyResponse {
  message: string;
  key: string; // ONLY returned once on regeneration
  warning: string;
}